stock_request_separate_pickings/models/stock_request_abstract.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2023 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class StockRequestAbstract(models.AbstractModel):
+    _inherit = "stock.request.abstract"
+
+    procurement_group_id = fields.Many2one(
+        copy=False,
+    )
+
+    @api.model
+    def create(self, vals):
+        res = super().create(vals)
+        if not res.procurement_group_id:
```

Comment:
@ps-tubtim As this break stock_request behavior, maybe a parameter to activate it (on company with res.config.settings) should be great

Reply:
@rousseldenis Thank you for your suggestion. I already add the **Allow Separate Picking** setting. Could you review this module again?